zellij-client/src/input_handler.rs

```diff
@@ -193,3 +242,45 @@ pub(crate) fn input_loop(
     )
     .handle_input();
 }
+
+struct StdinPoller {
+    poll: Poll,
+    events: Events,
+    timeout: time::Duration,
+}
+
+impl StdinPoller {
+    // use mio poll to check if stdin is readable without blocking
+    fn ready(&mut self) -> bool {
+        self.poll
+            .poll(&mut self.events, Some(self.timeout))
+            .expect("could not poll stdin for readiness");
+        for event in &self.events {
+            if event.token() == Token(0) && event.is_readable() {
+                return true;
+            }
+        }
+        false
+    }
+}
+
+impl Default for StdinPoller {
+    fn default() -> Self {
+        let stdin = 0;
+        let mut stdin_fd = SourceFd(&stdin);
+        let events = Events::with_capacity(128);
+        let poll = Poll::new().unwrap();
+        poll.registry()
+            .register(&mut stdin_fd, Token(0), Interest::READABLE)
+            .expect("could not create stdin poll");
+
+        // TODO: decide what timeout to use
+        let timeout = Duration::from_millis(50);
```

Comment:
I think this can even be less, 10ms?

Reply:
related to the comment just below, so it should be 10ms to have faster scrolling?